sdk/src/main/scala/com/horizen/network/SidechainNodeViewSynchronizer.scala

```diff
@@ -45,8 +56,160 @@ class SidechainNodeViewSynchronizer(networkControllerRef: ActorRef,
+        // parse all transactions and send them to node view holder
+        val parsed: Iterable[SidechainTypes#SCBT] = parseModifiers(requestedModifiers, serializer, remote)
+        viewHolderRef ! TransactionsFromRemote(parsed)
+
+      case Some(serializer: SparkzSerializer[SidechainBlock]@unchecked) =>
+        // parse all modifiers and put them to modifiers cache
+        val parsed: Iterable[SidechainBlock] = parseModifiers(requestedModifiers, serializer, remote)
+        val valid: Iterable[SidechainBlock] = parsed.filter(validateAndSetStatus(remote, _))
+        if (valid.nonEmpty) viewHolderRef ! ModifiersFromRemote[SidechainBlock](valid)
+
+      case _ =>
+        log.error(s"Undefined serializer for modifier of type $typeId")
+    }
+  }
+
+  /**
+   * Method copied as-is  from the parent class to be used by the previous
+   * TODO: this method can be removed once inherited method from Sparkz will be changed to protected visibility
+   *
+   * Parse modifiers using specified serializer, check that its id is equal to the declared one,
+   * penalize misbehaving peer for every incorrect modifier or additional bytes after the modifier,
+   * call deliveryTracker.onReceive() for every correct modifier to update its status
+   *
+   * @return collection of parsed modifiers
+   */
+  private def parseModifiers[M <: NodeViewModifier](modifiers: Map[ModifierId, Array[Byte]],
+                                                    serializer: SparkzSerializer[M],
+                                                    remote: ConnectedPeer): Iterable[M] = {
+    modifiers.flatMap { case (id, bytes) =>
+      val reader: VLQReader = new VLQByteBufferReader(ByteBuffer.wrap(bytes))
+
+      serializer.parseTry(reader) match {
+        case Success(mod) if id == mod.id =>
+          if (reader.remaining != 0) {
+            penalizeMisbehavingPeer(remote)
+            log.warn(s"Received additional bytes after block. Declared id ${encoder.encodeId(id)} from ${remote.toString}")
+          }
+          Some(mod)
+        case _ =>
+          // Penalize peer and do nothing - it will be switched to correct state on CheckDelivery
+          penalizeMisbehavingPeer(remote)
+          log.warn(s"Failed to parse modifier with declared id ${encoder.encodeId(id)} from ${remote.toString}")
+          None
+      }
+    }
+  }
+
+  /**
+   * Method copied from the parent class, with the addition of isReindexing check: we must avoid to penalize
+   * a peer if receiving a block during reindex
+   *
+   * Move `pmod` to `Invalid` if it is permanently invalid, to `Received` otherwise
+   */
+  @SuppressWarnings(Array("org.wartremover.warts.IsInstanceOf"))
+  private def validateAndSetStatus(remote: ConnectedPeer, pmod: SidechainBlock): Boolean = {
+    if (isReindexing) {
+      log.error("Got modifier while reindexing - will be discarded")
+      deliveryTracker.setReceived(pmod.id, remote)
```

Comment:
This is exactly to cover the case you listed here:  
- we requested previously a block
- started a reindex
- the block arrives: we don't have to apply it, but we mark it as received, otherwise the peer we are asking to will be penalized after an amount of time (see deliveryTracker.onStillWaiting )

SyntacticallySuccessfulModifier is only called by NVHS during processRemoteModifiers but there is already a isReindexing() there

Reply:
after discussion: considered as an issue. to be addressed